server/src/main/java/org/apache/druid/server/coordination/SegmentBootstrapper.java

```diff
@@ -261,9 +267,16 @@ private void loadSegmentsOnStartup() throws IOException
 
   /**
    * @return a list of bootstrap segments. When bootstrap segments cannot be found, an empty list is returned.
+   * The bootstrap segments returned are filtered by the broadcast datasources indicated by {@link DataSourceTaskIdHolder#getBroadcastDatasourceLoadingSpec()}
+   * if applicable.
    */
   private List<DataSegment> getBootstrapSegments()
   {
+    final BroadcastDatasourceLoadingSpec.Mode mode = datasourceHolder.getBroadcastDatasourceLoadingSpec().getMode();
```

Comment:
Can we use datasourceHolder.getLookupLoadingSpec() here and not add any new configs  ?

Reply:
related comment: https://github.com/apache/druid/pull/17027#discussion_r1753165605